### PR TITLE
Add compatibility.mdx

### PR DESCRIPTION
**Pull request by BrynCooke**
### Compatibility docs [PR #896](https://github.com/apollographql/router/pull/896)
Add a page about compatibility with Federation versions.

### CODE REVIEW

**Review by Geal — APPROVED** [review]

**Review by abernix — COMMENTED** [review]
This looks really great.   Since this diverges a bit from what we discussed with @StephenBarlow previously, let's just get explicit approval from him before merging.

File: docs/source/compatibility.mdx
Comment (StephenBarlow) [inline docs/source/compatibility.mdx:25]: We can remove `required` from all of these `tr`s

File: docs/source/compatibility.mdx
Comment (StephenBarlow) [inline docs/source/compatibility.mdx:14]: Don't need these classes on this particular table (they're specifically for lists of function parameters or object fields)

File: docs/source/compatibility.mdx
Comment (StephenBarlow) [inline docs/source/compatibility.mdx:3]: Cut description (it appears directly below the title and is redundant in this case)

File: docs/source/compatibility.mdx
Comment (StephenBarlow) [inline docs/source/compatibility.mdx:2]:
```suggestion
title: Federation version support in the Apollo Router
```

File: docs/source/config.json
Comment (StephenBarlow) [inline docs/source/config.json:42]: Except at the very top of the navigation, we try not to have single items that aren't categorized.

I recommend `Federation version support` instead of `Compatibility` and putting just below `"Migrating from @apollo/gateway"` above

**Review by StephenBarlow — APPROVED** [review]
Looks good with recommendations!

**Pull request closed without merging**
